### ISSUE TITLE
annual report 2020. due May 14

### ISSUE THREAD

**Issue by myrmoteras** [plazi/arcadia-project#144]
The annual report for Arcadia is due May 14, 2020

**Comment by punkish** [comment]
# YEAR TWO

1. Liberation of data from scholarly publications
    1.1. Improve online data quality control and correction tool;
        1.1.1. add better feedback facilities to TB search portal
        1.1.2. finalize TB search portal correction facilities
        1.1.3. Define what we out date
        1.1.4. Install a feedback mechanism for data issues	
    1.2. Set up ingestion of taxpub based articles; and
         1.2.1. already in place for Pensoft - only need to generalize
    1.3. Extend template based extraction of 70K additional treatments including those from the most relevant taxonomy journals
2. Infrastructure
    2.1. Migrate existing TreatmenBank data to Zenodo; and 
        2.1.1. This is dependent on 1.2.1
    2.2. Adjust the extraction process so new output goes to Zenodo
3. Interfaces, Discovery tools and APIs
    3.1. Create sample applications to demonstrate discovery and analytical capabilities of the API
4. BLR Web Presence
    4.1. Enhance the website with rich information discovery tools
    4.2. Brand and design the website
    4.3. Refine and test the UI
5. Outreach
    5.1. Provide a daily summary of new data liberated on Twitter and Facebook
    5.2. Attend at least three major conferences, two in Europe and one in the United States or elsewhere; and
    5.3. Publish a scholarly publication describing our work

**Comment by punkish** [comment]
Please look at each of the major activity groups above (numbered 1-5). Under each of them, please read through each of the tasks listed (numbered x.y.). For whichever applies to you, please fill out the following template.  (Make a new comment in this thread, copy the template, and fill it out.)

Please *do not* exceed the prescribed word limits. The report *has* to be short, and editing long texts down to the required length becomes very, very tedious. Provide URLs where needed (URLs don't count in the word limits).

***Note:** Task 1.1. has a list of example sub-tasks (starting with -). Modify them as needed, and create a sub-task similarly under other tasks, as required.*

# x.y. &lt;Task Name&gt;

## Summary
50 words or less.

**url**:

## Progress
50 words or less.

**url**:

**Comment by punkish** [comment]
# 3.1. API Development

## Summary
The version 2 of the Zenodeo API was released with query capabilities to the complete TreatmentBank dataset. The API was also made more performant and equipped with added capabilities for future development of more analytical insights into the data as well as its use by the users.

**url**: https://zenodeo.punkish.org/v2

## Progress
The Zenodeo API received a major version upgrade that exposed all the treatment-related data in the TreatmentBank to RESTful queries. The underlying data store and caching mechanism were tuned for high performance no matter how complex the query. The data store and the querying mechanism themselves were modified with self-tracking capabilities so every query would be recorded with its query count and performance. This lays the foundation for carrying out insightful analysis of the use of data.

**image**:

**Comment by punkish** [comment]
# 3.2. Ocellus, a sample application to demonstrate API capabilities

## Summary
The image discovery application Ocellus that was developed in the year 1 of the Arcadia Project also received an upgrade by way of a more refined user interface as well as laying the groundwork for further development of more granular querying capabilities.

**url**: https://ocellus.punkish.org/

## Progress
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus dolor felis, viverra sit amet felis a, aliquam molestie mi. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut non porttitor mauris, eu auctor augue. Suspendisse potenti. Sed tincidunt, ipsum a maximus ultrices, lacus orci sagittis odio, in varius metus sapien.

**image**: 
![Ocellus 1](https://user-images.githubusercontent.com/349242/80650961-444e2480-8a75-11ea-8444-594432e41bad.jpeg)

![Ocellus 2](https://user-images.githubusercontent.com/349242/80651011-5e880280-8a75-11ea-83ce-897f8d5921c9.jpeg)

**Comment by myrmoteras** [comment]
@tcatapano @slint @lnielsen  @gsautter @advocomplex @mguidoti @teodorgeorgiev @lyubomirpenev 

at the begin of this issue you find the description of how puneet and I plan to write this time the report (the copy of the last, too long report is [here](https://drive.google.com/open?id=10hYTVtNBz4b7exn5hqwhzWBaE3tRCw-o) to give you an idea how it will look like.

We would appreciate if you could return your parts by Tuesday 5.

**Comment by myrmoteras** [comment]
@gsautter 
here what we discussed and need: 
* stats: user stats for TreatmentBank
* stats: QC tool, eg how many articles processed, types of errors, time needed to fix, 
* developments since May 2019
* planned further developments
* new release of the programs in involved

**Comment by mguidoti** [comment]
Hi Donat,

Nothing on the 50% of n. sp./year goal? Lycophron, O3RT, training materials?

Thanks

**Comment by myrmoteras** [comment]
@mguidoti 
The stats should be a summary from May 16 2019 to May as close as possible to May 14 2020. I would also like to see the figures from September 1, 2019  to May 1, 2020; we should also show monthtly production, number of template based treatments, etc.

* stats for treatment imports,  
* number of templates
* prediction for the production year Sept 1 2019 - August 31, 2020.

We might want to discuss lower of higher number, as what changes we plan to get to 200,000K treatments in y3 (tables, etc.)

**Comment by mguidoti** [comment]
Ok, I'll do my best!

Thanks

**Comment by punkish** [comment]
> Ok, I'll do my best!
> 


awesome. Numbers showing productivity increases are very important. Would be lovely if you can create a couple of charts (see the option of images). This in itself can also be a standalone blog post on the Plazi website.

**Comment by mguidoti** [comment]
Yes, that's exactly what I've in mind. I'll definitely do my best. Time is a bit strict, as I spent most of my day micromanaging the office and answering question, but I'll do my best.

**Comment by myrmoteras** [comment]
3. Describe Synospecies as an application based on our data https://synospecies.plazi.org/

**Comment by punkish** [comment]
> 1. Describe Synospecies as an application based on our data https://synospecies.plazi.org/

This looks like it belongs under Activity 3, so I have added it to the list above as 3.3. (see https://github.com/plazi/arcadia-project/issues/144#issuecomment-621471556).

***Note to everyone:** As stated in the linked post, please feel free to add new activities, but they **must** be at the 'x.y.' level. Please **do not** create a new top-level activity*

@myrmoteras Please assign someone to write up this task (probably Reto) as it won't write itself.

**Comment by mguidoti** [comment]
Hi, 

Here's my input on what was asked from me. I hope you find this useful. I followed the 50-word limit and on the productivity report I focused on what Donat personally told me to do. If any clarification is needed, please, just ping me.

### Lycophron
> 49 words
> 
Lycophron is a Python script develop to batch upload publications to Zenodo from a spreadsheet containing their metadata. After the last Arcadia Sprint Meeting the Zenodo team asked Plazi to merge Lycophron with their Batch Uploader. This is an ongoing project that will be finished in the upcoming months.

Repo: https://github.com/plazi/lycophron

### O3RT
> 48 words
> 
O3RT is a Open Refine Reconciliation tool that queries Refindit retrieving the most likely DOIs for publications based on their title. This was created to avoid DOI publication when batch uploading publications to Zenodo. The user then can manually select or evaluate the choices made by the program.

Repo: https://github.com/plazi/O3RT

### Learning Sources
> 47 words
> 
Screecasts describing the process of treatment extraction from a single article were recorded with subtitles in English available, these under proof-reading. Screencasts explaining the template creation process and the quality check are planned. A 8-hour long minicourse was developed, with 20h and 40h versions under currently development.

Currently Available Screencasts: https://www.youtube.com/playlist?list=PLFbvkmnvLdUdGmmn8SR4xyRRxulvVu7BE

**Comment by mguidoti** [comment]
## Plazi@Poa Productivity Report
> Last Update: 2020-05-09

### TL;DR
- Current production rate: **~4h/template**
- Current production rate (number of pages/hour): **79 pages/h**

### Disclaimers

- **GitHub Repos**: [mguidoti/TBtoNotion](https://github.com/mguidoti/tbtonotion) (hits TB API collecting data on treatments and bibRefs from the articles processed), [mguidoti/arcadia_report_y2](https://github.com/mguidoti/arcadia_report_y2) (Jupyter Notebooks used to analyzed the data).
- I can transfer the ownership of these repos if you want. Right now they are closed, because I believe they have some sensible information, but I granted access to you. **_I'm still commiting my latest edits._**
- I exported all dataframes to `*.xlsx` and will be providing the links below.
- I did not work on charts because Donat said we wouldn't have space for it in the report.
- I tried to be as brief as I can to explain this complex information. 


### Raw data

* [Plazi-at-Poa_Templates_2020-05.csv](https://github.com/mguidoti/arcadia_report_y2/blob/master/data/Plazi-at-Poa_Templates_2020-05.csv)
* [Plazi-at-Poa_Article_Extractions_2020-05.csv](https://github.com/mguidoti/arcadia_report_y2/blob/master/data/Plazi-at-Poa_Article_Extractions_2020-05.csv)

As I explained during my presentation in last February, we're using an app called Notion.so to keep track of the production rate. Then, I run a script I coded to download some data from the TB API to complete the database (TBtoNotion). These are the data fields I collect from TB API:

- ContTreatCount
- ContTreatCitCount
- ContMatCitCount
- ContFigCount
- ContTabCount
- ContBibRefCount
- MatSpecimenCount
- ContPageCount
- DocUploadDate

### Teams Time Management (Main focus)


| Person | Nov/2019 | Dec/2019 | jan/2020 | feb/2020 | mar/2020 | apr/2020 | may/2020 |
| -------- | -------- | -------- | -------- | -------- | -------- | -------- | -------- |
| Tatiana | training | training | templates | learning mat | Acta/Corona Viruses Project | Corona Viruses Project | Corona Viruses Project |
| Carolina | training | training | templates | NI | NI/RSZ | Zootaxa new | EJT backlog |
| Valdenar | training | learning material/training | learning material | learning material | Linzer backlog | EJT new/Linzer backlog | EJT new and backlog/Linzer backlog |
| Felipe | - | - | training/templates | IM/ZS/templates | IM/ZS/templates | templates | ZK/templates |

* IM = Insecta Mundi backlog
* ZS = Zoosystema backlog
* ZK = ZooKeys backlog
* Acta = Acta Entomologica Pragae
* NI = Neotropical Ichthyology
* RSZ = Revue Suisse de Zoology



### On Templates

* Current production rate: **~4h/template**

It dropped from ~10h/template to ~4h/template last month due to a `express template creation workflow` we established. The big difference is the reduction of the amount of papers on the testing sample, and the amount of time we initially spend investigating how suitable the journals are for templating. However, we still document the whole process, which culminates in a guide highlighting the biggest issues and challenges on working with a particular journal.

**At this point, I don't see how we could improve this production rate. I think we hit a limit considering the quality/time spent trade-off.**

Plus, GGI improved greatly from our direct feedback to Guido and his work in the past few months. Sometimes we had to wait a little bit until we could go back and work with a given journal, which in my opinion was expected as journals vary a lot in terms of layout and no one before us put this amount of time into templating. We compiled a list of these improvements caused/motivated by our input [here](https://github.com/plazi/plazi-at-poa/wiki/GGI-Development:-Features-implemented).

#### Data Summarized
- [Production rate by Months of Operation](https://github.com/mguidoti/arcadia_report_y2/blob/master/results/Plazi-at-Poa_template_production_rate.xlsx)
- [Production rate by Months and People/Worker](https://github.com/mguidoti/arcadia_report_y2/blob/master/results/Plazi-at-Poa_template_production_rate_by_people.xlsx)


### On Article Processing

* Current production rate (number of pages/hour): **79 pages/h**

Here I summarized data considering `dedicated hours`, `number of pages processed` and `number of treatments produced`. Data might not meet TB's for several reasons - to name one, for instance, the latest ZooKeys uploaded by the office was downloaded by Guido to the server, which introduced a given `first upload date` that is different than the `final date` used by us in Notion, or, sometimes we change this date for the `Concluded` articles to fit additional edits we had to make (e.g., errors that we notice on GBIF).

From these, I think the only production rate statistic that makes sense is `number of pages in an hour` because articles and treatments vary in number of pages, but pages are an entity on its own. Yes, they can vary in complexity but still it's the most comparable entity that we have. Thus, this is the production rate I'm presenting here, although the ones based on treatments and articles were included in the data summary files below.

In April we adopted a new level of granularity, which we call `aleluia-level`. It's the lowest level of granularity that the office operated since we started. Thus, the order of levels adopted thus far, from highest to lowest, is: 

1. High-level
2. Low-level
3. Aleluia-level 

Based on this, we improved from 17 pages/h in March to 47 pages/h in April (aleluia-level started) to 79 pages/h in May (only Aleluia-level). You can check how each of the levels contributed to the production rate in one of the data summaries below.

One point that I still have to make is the fact that it's not only the granularity that impacts production rate. These are important factors that impact the production rate as well:

- journal template consistency;
- template quality;

This means that a low-level granularity on a inconsistent journal can take more time than a high-level granularity in a 'good' journal, like EJT (which is, by the way, the only journal being processed in high-level since March).

Granularity levels and changes on focus and how we operate were extensively discussed with Donat, and we followed every decision made in these conference calls, as well as last-minute requests and changes.

**I also think we hit the best trade off between data quality/production rate and we won't be able to improve much further**. Maybe with time and experience we will get a bit faster. We guarantee that there are no missing neither broken treatments, especially new species, at the current workflow and production rate. The same can't be affirmed for Zootaxa, for instance.

#### Data Summarized
- [Production rate by Months of Operation](https://github.com/mguidoti/arcadia_report_y2/blob/master/results/Plazi-at-Poa_article_production_rate.xlsx)
- [Production rate by Months and Annotation/Granularity level](https://github.com/mguidoti/arcadia_report_y2/blob/master/results/Plazi-at-Poa_article_production_rate_by_annotation.xlsx)
- [Production rate by Months and People and Annotation/Granularity level](https://github.com/mguidoti/arcadia_report_y2/blob/master/results/Plazi-at-Poa_article_production_rate_by_people_and_annotation.xlsx)

**Comment by myrmoteras** [comment]
@mguidoti thanks for the report and all the work by you and your team.

lets discuss it overall once we submitted the report.

In Arcadia, the reporting unit is **number of treatments**, **number of new species liberated**, and **number of templates**. For this reason, you can't change the reporting system to number of pages only, and I would like to ask for these numbers. I made an effort [here](https://github.com/plazi/Plazi-Communications/wiki/Statistics#basic-data-arcadia-may-2019---april-2020-1-2).

What did you do in regards of teaching? how many screencasts, other teaching materials we have?

It would be helpful if the list of treatments is accessible on Plazi communications github, and with the proper name of the journal, so to link it with the production statistics in plazi stats, eg [Journal of Paleontology](http://tb.plazi.org/GgServer/dioStats/stats?outputFields=bib.year+bib.source&groupingFields=bib.year+bib.source&FP-bib.source=%22Journal%20of%20Paleontology%22&format=HTML).

How do you define a finished template? Shouldn't it be that it successfully can process a batch of XX articles to the aleluia level, and then we could use this stats that it works?

Based on the 79 pages per hour, that translates in 39 treatments per hour (700K pages, 38K treatments). The process 250K treatments in year 3, we need  6,410 hours. using 1890 working hours per year, this would need 3,4 persons dedicated for this work.

Regarding graphics: Yes, we can't include more than one graphic. But for internal communications that is helpful nevetheless and should be considered.

Please also consider what data we regarding the production we can move to https://github.com/plazi/Plazi-Communications/wiki/Statistics so we have one central place with the figures, may including figures.

**Comment by myrmoteras** [comment]
@mguidoti Regarding Lycophron: we mentioned this already last year. in the report. Where do we stand with the import of taxodros that we planned to upload in y2?

**Comment by mguidoti** [comment]
Hi Donat,

> In Arcadia, the reporting unit is number of treatments, number of new species liberated, and number of templates. For this reason, you can't change the reporting system to number of pages only, and I would like to ask for these numbers. I made an effort here.

### On templates
The number of templates was given already, by Skype. **It's 24 and [here's the list](https://github.com/plazi/plazi-at-poa/wiki/Templates#currently-available-templates)**. This was also added [here](https://github.com/plazi/Plazi-Communications/wiki/POA), as you asked.

### On treatments

- **Number of treatments/h: 28.**

The number of treatments and species liberated you said yourself that you would get this from TB as TB has data coming from other sources too (Pensoft, you, Jeremy). What I'm offering here is a way of predicting what we will accomplish in the upcoming months (say, until August).

Number of treatments doesn't work for these estimates. One treatment can have a single line (catalogs, lists of distributional records) or many pages (full description). The time we spend, then, varies in orders of magnitude. Number of new species also makes no sense to be used in actual estimates as we can't predict the amount of new species that will exist in a given PDF. And we don't manually track the number of new species, we rely on TB. This is my opinion. But the treatments/h rate, even if doesn't really makes much sense on my humble opinion, was included in the shared data summary files from my previous post. Here they were included too, to facilitate.

Small note: we went from 11 treat/h to 27/h when we started the aleluia-level of granularity.

Plus, I'm not changing the reporting system, I'm showing an educated opinion based on actual production data after months of this operation. This is the first time we have this. When you guys wrote the project this wasn't present. Data can't be completely ignore, specially when it tells a different story - that's what the scientist on me tells me to say and that's why I'm highlighting the right things, on my opinion. As said previously, the numbers you need and want for the report you said you would get from TB, which makes sense. 

Finally, always keep in mind that there are some differences between the absolute number of treatments on our data summaries and the one you get from TB. **This is normal and expected** by a platoon of reasons, some cited on my original post here. 

Let me know if this helps. If not, tell me what's missing again and I'll do my best.

**Comment by mguidoti** [comment]
> What did you do in regards of teaching? how many screencasts, other teaching materials we have?

42. We did 42 screencasts, covering individual processing of articles only at this point. [On this link](https://www.youtube.com/playlist?list=PLFbvkmnvLdUdGmmn8SR4xyRRxulvVu7BE) we can see the number of videos too.

We also produced a list of commented steps for both individual processing and template making. This is intended to be part of the updated version of the manual, a halted project for now, due to all the other priorities that we have both individually and as an office. This, again, was not a decision taken by us, but it was discussed.

The courses production also halted. As you can see, we changed the focus of Valdenar and Tatiana in feb/2020 to other things. We have a draft of an outline for the 40h course, by Tatiana, waiting my and ultimately yours review.

I didn't specify all of these details on that provided paragraph in an attemptto follow the 50 word limit Puneet told me to.

**Comment by mguidoti** [comment]
> It would be helpful if the list of treatments is accessible on Plazi communications github, and with the proper name of the journal, so to link it with the production statistics in plazi stats, eg Journal of Paleontology.

Ok, I can definitely add a list of all template-based processing done by the office there. I just wrote on my task list. 

## On journal name problems
We manually fix the journal names (and change the value on the template) to normalize this as soon as we notice an error. 

But it was agreed that we would run something on the server to catch and perhaps fix this automatically in the future. It was said that we would worry about it after the project regarding authors names as we would learn a lot from it, and we could apply that experience to deal with this particular problem.

**Comment by mguidoti** [comment]
> How do you define a finished template? Shouldn't it be that it successfully can process a batch of XX articles to the aleluia level, and then we could use this stats that it works?

A finished template is one that produces IMFs that takes less than 15 minutes to be fixed **to the aleluia level**, on average. That's the golden rule we have been applying to define what's a finished template.

From our experience, we can't simply blindly trust a batch process. Looking at Zootaxa papers done this way you can see that even when the dwc file is produced, there might be not one but lots of treatments missed altogether inside that paper, including treatments about new species. This happens for several reasons and none of them that can be predicted and worked by Guido - journals simply vary too much, even from paper to paper. Templates can't predict all of these variations. It's just a blueprint.

The right way to process articles, **guaranteeing a minimum level of data quality**, is to open every and each IMF, correct errors including the ones from QC (blockers + one critical regarding treatment) and only then uploading. On automation, we could batch process on the server. This would save some time, but I would never get rid of a human checking it before uploading the data. Especially when we start issuing DOIs for each and every treatment on the fly.

Again, that's what our experience from the front lines teaches us.

**Comment by mguidoti** [comment]
> Based on the 79 pages per hour, that translates in 39 treatments per hour (700K pages, 38K treatments). The process 250K treatments in year 3, we need 6,410 hours. using 1890 working hours per year, this would need 3,4 persons dedicated for this work.

That's exactly why the treatments/h rate doesn't work. You overestimated according to out data. Our data shows a 28 treat/h production rate, not 39. You can't do the pages/treat, this doesn't mean much when treatments can have 1 paragraph or several pages.

Following your reasoning and the actual data (28 treat/h), we need 8,928 hours. Considering the 1890 working hours per year, we would need 4.7 people. Right now we have 4 people in the office, but 1 is allocated to work on the Corona Viruses Project, and the other is basically doing templates only. **The time to produce enough templates to have enough papers to process and get treatments also needs to be into your estimates.** This means that we have **only 2** exclusively focused on extracting treatments and this will last until the next last minute request.

We are short on human power when we look at the real data, not pages/treat. 

That's why I'm pushing forward our statistics. They came from real experience in the processing facility.

**Comment by mguidoti** [comment]
> Regarding graphics: Yes, we can't include more than one graphic. But for internal communications that is helpful nevetheless and should be considered.

Ok, will do.

**Comment by myrmoteras** [comment]
> Hi Donat,
> 
> > In Arcadia, the reporting unit is number of treatments, number of new species liberated, and number of templates. For this reason, you can't change the reporting system to number of pages only, and I would like to ask for these numbers. I made an effort here.
> 
> ### On templates
> The number of templates was given already, by Skype. **It's 24 and [here's the list](https://github.com/plazi/plazi-at-poa/wiki/Templates#currently-available-templates)**. This was also added [here](https://github.com/plazi/Plazi-Communications/wiki/POA), as you asked.

that means the [figure 47](https://github.com/mguidoti/arcadia_report_y2/blob/master/data/Plazi-at-Poa_Templates_2020-05.csv) is the number of templates, for 24 different journals?

**Comment by mguidoti** [comment]
> > Hi Donat,
> > > In Arcadia, the reporting unit is number of treatments, number of new species liberated, and number of templates. For this reason, you can't change the reporting system to number of pages only, and I would like to ask for these numbers. I made an effort here.
> > 
> > 
> > ### On templates
> > The number of templates was given already, by Skype. **It's 24 and [here's the list](https://github.com/plazi/plazi-at-poa/wiki/Templates#currently-available-templates)**. This was also added [here](https://github.com/plazi/Plazi-Communications/wiki/POA), as you asked.
> 
> that means the [figure 47](https://github.com/mguidoti/arcadia_report_y2/blob/master/data/Plazi-at-Poa_Templates_2020-05.csv) is the number of templates, for 24 different journals?

Yes, and some are not `Status=Completed`. This spreadsheet is the raw data.

Here you can see three entries for the same journal, for different purposes. We do like this to track the time for each template we produce, regardless the journal.
![image](https://user-images.githubusercontent.com/44419136/81501117-f1fad800-92ac-11ea-9aa8-6deab5f695aa.png)

**Comment by mguidoti** [comment]
> @mguidoti Regarding Lycophron: we mentioned this already last year. in the report. Where do we stand with the import of taxodros that we planned to upload in y2?

You haven't mentioned the fact that Lycophron is not a Plazi tool anymore, but it will be the official Zenodo Uploader, developed by the two combined. It shows acceptance of the idea by a CERN initiative. I think it's relevant. That's what I wrote in my 50-word long paragraph.

Taxodros is different than Lycophron, especially now that it outgrow the project. Taxodros project, however, will use Lycophron. 

For taxodros, I've to:
- Test O3RT, generating a confidence stats on the tool;
- Work with Alex on new Lycophron;
- Run both tools with TaxoDros database.

Why I haven't done this yet: office management, corona project, CBZ, fabricius types, doi registration, organizational tasks, BLR, etc.

Everything that pops up is urgent, basically, and I've to stop what I'm doing and work on that. I planned and scheduled with Zenodo team to work on Lycophron this month, after the report (on my mind). But it seems that you will want to change the office workflow again, so I'm assuming that we will put lots of hours into discussing or passing the new instructions to the people, while organizing the workflow in the current tools and so on. I'm looking forward to discuss this on the light of the data from the processing facility.

I hope this answers. If you need anything else, please, just ping me.

Cheers,

**Comment by myrmoteras** [comment]
@gsautter where are the repository for the code of the quality control  and the template tools? We need to provide this link.

**Comment by gsautter** [comment]
QC: https://github.com/gsautter/goldengate-qualitycontrol/ (with base classes in https://github.com/gsautter/idaho-core and https://github.com/gsautter/idaho-imagemarkup , the document listing for the QC Tool in https://github.com/gsautter/goldengate-server-imagemarkup/ , and the error stats a currently a custom implementation due to a few dependency issues I want to keep out of our main repos)

Template Tools: https://github.com/gsautter/goldengate-imagine-plugins/ (with base classes being spread out between https://github.com/gsautter/idaho-extensions and https://github.com/gsautter/idaho-imagemarkup )
